aten/src/ATen/cpu/vec256/vec256_complex_double_no_vec.h

```diff
@@ -0,0 +1,328 @@
+#pragma once
+
+#include <ATen/cpu/vec256/intrinsics.h>
+#include <ATen/cpu/vec256/vec256_base.h>
+#if defined(__AVX__) && !defined(_MSC_VER)
+#include <sleef.h>
+#endif
+
+namespace at {
+namespace vec256 {
+// See Note [Acceptable use of anonymous namespace in header]
+namespace {
+
+template <> class Vec256<std::complex<double>> {
+private:
+  std::complex<double> values[2] = {0};
+public:
+  using value_type = std::complex<double>;
+  static constexpr int size() {
+    return 2;
+  }
+  Vec256() {}
+  Vec256(std::complex<double> val) {
+    for (int i = 0; i != size(); i++) {
+      values[i] = val;
+    }
+  }
+  Vec256(std::complex<double> val1, std::complex<double> val2) {
+    values[0] = val1;
+    values[1] = val2;
+  }
+  template <int64_t mask>
+  static Vec256<std::complex<double>> blend(const Vec256<std::complex<double>>& a, const Vec256<std::complex<double>>& b) {
+    int64_t mask_ = mask;
+    Vec256 vec;
+    for (int i = 0; i < size(); i++) {
+      if (mask & 0x01) {
+        vec[i] = b[i];
+      } else {
+        vec[i] = a[i];
+      }
+      mask_ = mask_ >> 1;
+    }
+    return vec;
+  }
+  static Vec256<std::complex<double>> blendv(const Vec256<std::complex<double>>& a, const Vec256<std::complex<double>>& b,
+                               const Vec256<std::complex<double>>& mask) {
+    Vec256 vec;
+    int64_t buffer[size()];
+    mask.store(buffer);
+    for (int i = 0; i < size(); i++) {
+      if (buffer[i] & 0x01)
+       {
+        vec[i] = b[i];
+      } else {
+        vec[i] = a[i];
+      }
+    }
+    return vec;
+  }
+  static Vec256<std::complex<double>> arange(std::complex<double> base = 0., std::complex<double> step = 1.) {
+    return Vec256(base, base + step);
```

Comment:
This is "simplified" from the loop (and thus diverges from base), but I don't really see the utility of unrolling it as you've done here; since `size()` is constexpr won't it get unrolled anyway?

Reply:
Yeah, I appreciate the feedback on that. 

I haven't thought very much about the implementations in vec256_complex... yet.  I just put stuff there so I would stop copy and pasting the kernels for complex ops. I really need to take a fresh look at these implementations. Does that sound ok?

Reply:
Yes, I think taking a closer look is a good idea. I (and others) can answer questions about what the classes are doing. Based on my code review, I think you can directly reuse base for complex, with two additions:

1. You add a way of disabling methods which don't work for complex
2. We do some very minor adjustments to a few functions (like `abs`) to make them more complex friendly

If you need to unblock, I'm even willing to accept a copy of base for complex (in the short term). But I'm not willing to have two copies for float and double.